clevertap-core/src/main/java/com/clevertap/android/sdk/LocalDataStore.java

```diff
@@ -54,13 +54,15 @@ public class LocalDataStore {
 
     private final String eventNamespace = "local_events";
 
+    private final DeviceInfo deviceInfo;
```

Comment:
we need to check any cases if device info async block is not completed and we fetch deviceId premtively.

Reply:
Yess. Added to testing scope todos